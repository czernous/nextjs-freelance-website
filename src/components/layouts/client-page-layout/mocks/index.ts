import { IClientPageLayoutProps } from '../../../../interfaces';

export const clientPageMock: IClientPageLayoutProps = {
  pageTitle: 'About',
  meta: {
    metaDescription:
      'This is a dummy description designed for testing purposes',
    metaKeywords: 'some meta keywords,testing,typescript,blog',
    openGraph: {
      title: 'This title will be shown on social media',
      description: 'This description will be shown on social media',
      imageUrl: '',
      type: '',
      url: '',
    },
  },
};
