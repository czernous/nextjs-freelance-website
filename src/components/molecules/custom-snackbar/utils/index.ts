import { ICustomSnackbarProps } from '@src/interfaces/components-props';
import { SetStateAction } from 'react';

export const updateSnackbarProps = async (
  r: Response,
  callback: (value: SetStateAction<ICustomSnackbarProps | null>) => void,
) => {
  if (!r) return;
  const severity =
    r.status === 200 || r.status === 204 || r.status === 201
      ? 'success'
      : 'error';
  const hasJson = r.headers.get('Content-Type')?.includes('application/json');
  /* istanbul ignore next */
  const statusMessageText = async (r: Response) => {
    switch (r.status) {
      case 204:
        return 'Successfully saved data';
      case 201:
        return 'Successfully created';
      default:
        return await r.text();
    }
  };

  let statusMsg;

  try {
    statusMsg =
      hasJson && r.headers.get('Method') === 'PUT'
        ? await r.json()
        : {
            message: await statusMessageText(r),
          };
  } catch (error) {
    /* istanbul ignore next */
    statusMsg = hasJson
      ? {
          message:
            statusMsg.message ||
            (r.status === 200
              ? 'Successfully saved data'
              : 'Error saving data'),
        }
      : { message: 'Error reading response as text' };
  }

  callback({
    severity,
    text: statusMsg?.message ?? statusMsg,
  });
};
