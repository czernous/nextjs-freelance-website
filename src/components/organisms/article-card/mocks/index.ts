import { IArticleCard, IImage } from '../../../../interfaces';
import { base64ImgSmall, base64ImgMedium, base64ImgLarge } from './images.mock';

const imageMock: IImage = {
  publicId: 'fakeurl.com/asd',
  version: 1,
  signature: 'fakeSignature',
  width: 1000,
  height: 666,
  format: 'jpg',
  resourceType: 'fake resource',
  bytes: 111,
  type: '',
  url: base64ImgLarge,
  secureUrl: base64ImgLarge,
  responsiveUrls: [
    { width: 1000, url: base64ImgLarge },
    { width: 700, url: base64ImgMedium },
    { width: 500, url: base64ImgSmall },
  ],
  thumbnailUrl: base64ImgSmall,
  blurredImageUrl: base64ImgSmall,
  path: 'fakeurl',
  id: {
    timestamp: 111,
    machine: 111,
    pid: -999,
    increment: 111,
    creationTime: '2022-06-25T15:44:40.411Z',
  },
  _id: '12345',
  createdAt: '2022-06-25T15:44:40.411Z',
  updatedAt: '2022-06-25T15:44:40.411Z',
  altText: 'mock image',
};

export const cardMock: IArticleCard = {
  title: 'Article title',
  unoptimized: true,
  description:
    'Some random article short summary. This will be taken from the article itself or prefilled when the article is posted',
  ctaText: 'Read more',
  ctaUrl: 'https//google.com',
  image: imageMock,
};
