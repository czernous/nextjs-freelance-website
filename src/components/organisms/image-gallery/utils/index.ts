import { IImage } from '@src/interfaces';
import { Dispatch, SetStateAction } from 'react';

/* istanbul ignore next */
export const getImages = async (): Promise<IImage[]> => {
  const imageResponse = await fetch(
    `${new URL(
      '/api/blog-data',
      window.location.origin,
    )}?url=/images&method=GET`,
  );

  const r = await imageResponse?.json();

  return r?.data;
};

/* istanbul ignore next */
export const handleGalleryOpen = async (
  setImages: Dispatch<SetStateAction<IImage[] | null>>,
) => {
  const imageResponse = await fetch(
    `${new URL(
      '/api/blog-data',
      window.location.origin,
    )}?url=/images&method=GET`,
  );

  const imagesData = await imageResponse?.json();

  setImages(imagesData?.data);
};

/* istanbul ignore next */
export const handleImageDelete = async (imageId: string) => {
  const imageDeleteResponse = await fetch(
    `${new URL(
      '/api/blog-data',
      window.location.origin,
    )}?url=/images/${imageId}&method=DELETE`,
  );

  if (imageDeleteResponse?.status === 204) {
    // TODO: add toast notification upon delete
  }

  if (imageDeleteResponse?.status === 404) {
    alert(
      'This image does not exist, you have probably deleted it already. Please reopen the gallery',
    );

    // TODO: add toast notification upon delete
  }
};
