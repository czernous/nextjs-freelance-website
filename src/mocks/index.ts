export * from './page-data';
export * from './posts';
export * from './paginated-posts';
