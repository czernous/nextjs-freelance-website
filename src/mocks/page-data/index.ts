import { IHomePage, IServicesPage } from '@src/interfaces';

export const homePageMock: IHomePage = {
  pageFields: {
    ctaBtnHref: 'https://google.com',
    ctaBtnText: 'Get stuff',
    ctaHeadline: 'Are you looking to get awesome stuff??',
    ctaSubheadline: 'We have the most awesome stuff on the internet!',
  },
  slug: 'home',
  meta: {
    metaDescription: 'Test',
    metaKeywords: 'test',
    openGraph: {
      title: 'Test',
      description: 'test',
      imageUrl: 'http://g.com',
    },
  },

  updatedAt: '2023-02-16T15:47:25.559Z',
};

export const servicesPageMock: IServicesPage = {
  pageFields: {
    content:
      '<h2>Service #1</h2><p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur quis nunc libero. Nullam tincidunt lectus ut maximus dignissim. Mauris et risus in metus ornare convallis. Nullam in orci leo. Aenean sit amet arcu vitae ipsum mollis blandit. Pellentesque pellentesque sapien mollis, pharetra magna sit amet, varius eros. Donec eu dui ullamcorper ligula rutrum tincidunt. Sed eu venenatis mi. Nullam rutrum sagittis tortor, id feugiat dolor commodo a. Nullam venenatis rutrum elementum. Vestibulum vehicula lectus in nisl porta scelerisque at id metus. Sed porttitor purus eu cursus porta. Donec suscipit semper ligula, ut eleifend nunc commodo ut. Aenean sed mauris molestie, gravida nisi et, lobortis sapien. Nullam et massa mi. </p> <p>Nulla a sem sit amet nisl interdum tincidunt. Fusce pellentesque nec risus et lacinia. Donec dignissim enim ac vehicula malesuada. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam at varius augue. Phasellus varius, sem ac posuere mattis, augue turpis sollicitudin arcu, ut tincidunt arcu tellus vel sapien. Nulla sagittis auctor velit, ut tristique nulla sollicitudin et. Duis ac purus efficitur, congue purus egestas, ultricies quam. Nulla tincidunt nibh risus, vel semper lacus consectetur quis. Aenean luctus enim in justo rhoncus dignissim. Donec pellentesque fermentum sapien vitae aliquam.</p><h2>Service #1</h2><p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur quis nunc libero. Nullam tincidunt lectus ut maximus dignissim. Mauris et risus in metus ornare convallis. Nullam in orci leo. Aenean sit amet arcu vitae ipsum mollis blandit. Pellentesque pellentesque sapien mollis, pharetra magna sit amet, varius eros. Donec eu dui ullamcorper ligula rutrum tincidunt. Sed eu venenatis mi. Nullam rutrum sagittis tortor, id feugiat dolor commodo a. Nullam venenatis rutrum elementum. Vestibulum vehicula lectus in nisl porta scelerisque at id metus. Sed porttitor purus eu cursus porta. Donec suscipit semper ligula, ut eleifend nunc commodo ut. Aenean sed mauris molestie, gravida nisi et, lobortis sapien. Nullam et massa mi. </p> <p>Nulla a sem sit amet nisl interdum tincidunt. Fusce pellentesque nec risus et lacinia. Donec dignissim enim ac vehicula malesuada. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam at varius augue. Phasellus varius, sem ac posuere mattis, augue turpis sollicitudin arcu, ut tincidunt arcu tellus vel sapien. Nulla sagittis auctor velit, ut tristique nulla sollicitudin et. Duis ac purus efficitur, congue purus egestas, ultricies quam. Nulla tincidunt nibh risus, vel semper lacus consectetur quis. Aenean luctus enim in justo rhoncus dignissim. Donec pellentesque fermentum sapien vitae aliquam.</p>',
  },
  slug: 'services',
  meta: {
    metaDescription: 'string',
    metaKeywords: 'string',
    openGraph: {
      title: 'string',
      description: 'string',
      imageUrl: 'string',
    },
  },

  updatedAt: '2023-02-16T07:05:35.214Z',
};
