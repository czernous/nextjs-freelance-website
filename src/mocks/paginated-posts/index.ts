import { posts } from '../posts';

export const mockPagedData = {
  data: posts.slice(0, 10),
  hasPagination: true,
  page: 1,
  pageSize: 10,
  totalPages: Math.ceil(posts.length / 10),
  totalDocuments: posts.length,
};
