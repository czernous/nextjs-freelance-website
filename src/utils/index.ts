export * from './data-fetching';
export * from './testing';
