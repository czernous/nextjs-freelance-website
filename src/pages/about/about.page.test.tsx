import { render, RenderResult, screen } from '@testing-library/react';
import { axe, toHaveNoViolations } from 'jest-axe';
import '@testing-library/jest-dom';
import About from './index.page';
import { act } from 'react-dom/test-utils';
import React from 'react';
import { mockNextRouter } from '../../utils';
import { aboutPageMock } from '../../mocks';

expect.extend(toHaveNoViolations);

jest.mock('next/config', () => () => ({
  serverRuntimeConfig: {
    PROJECT_ROOT: __dirname,
  },
}));

let component: RenderResult;

beforeEach(async () => {
  mockNextRouter();
  component = render(<About data={aboutPageMock} />);
});

describe('About', () => {
  it('renders a content (p - description)', () => {
    const p = screen.getByTestId('paragraph');
    expect(p).toBeInTheDocument();
  });

  it('renders homepage unchanged', () => {
    const { container } = component;
    expect(container).toMatchSnapshot();
  });

  it('has no axe violations', async () => {
    const { container } = component;
    await act(async () => {
      expect(await axe(container)).toHaveNoViolations();
    });
  });
});
