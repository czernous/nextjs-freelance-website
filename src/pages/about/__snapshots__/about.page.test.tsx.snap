// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders homepage unchanged 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      class="image"
    >
      <img
        alt="photo of blog's author"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=https%3A%2F%2Fexample.com%2Ftest.jpg&w=3840&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Ftest.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fexample.com%2Ftest.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fexample.com%2Ftest.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fexample.com%2Ftest.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fexample.com%2Ftest.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fexample.com%2Ftest.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fexample.com%2Ftest.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fexample.com%2Ftest.jpg&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
    </div>
    <div
      class="content"
    >
      <p
        data-testid="paragraph"
      >
        This is an about page
      </p>
    </div>
  </div>
</div>
`;
